### DISCUSSION TITLE
Revalidation on yup's .when() condition change

### DISCUSSION THREAD

**Discussion by chornbec** [react-hook-form/react-hook-form#1679] [General]
Hello! I'm not sure I'm right on this.
I'm using yup as a validation scheme. Yup has a .when( 'inputName', { is: 'someValue', then: yup.string().required() } ) feature, but when the inputName is no longer 'someValue' the error message is still shown. I'm guessing it's because react-hook-form only triggers a watch() of some sort on the required input name... at least until the validation is satisfied.

In this codesandbox, if the lastName is "hello", the firstName is required, but when lastName is not 'hello', error message still exists. Maybe something I have to do in RHF's validationResolver?
[Codesandbox](https://codesandbox.io/s/react-hook-form-validationschema-v0wq4?file=/src/index.js)

**Comment by bluebill1049** [comment]
Is this depend field validation? React Hook Form focus at a input at a time. you will need to `triggerValidation` (`trigger` in V6) for other deepened fields.

**Reply by chornbec** [reply]
It's for fields that depend on other field values. Yup seems to keep track of the field values inside of .when() on every onChange event until that specific validation is true. I want to avoid using triggerValidation on every input's onChange, unless the error object is not empty.

**Reply by bluebill1049** [reply]
That's exactly what I meant above, Yup `validate` against the entire form values, RHF focus on single input at a time to provide better perf, however, you have the API to `trigger` dependent validation.

**Reply by bluebill1049** [reply]
we can trigger the entire form validation all the time, and let you to manage each field's dirty/touched and show error message accordingly,  but then why not just use controlled? it defeats the purpose of this lib.

**Reply by chornbec** [reply]
oh I see, thanks! I figured I'd have to manually triggerValidation for this specific yup schema. I tried putting triggerValidation in a useEffect(..., [errors]) whenever errors obj is not blank, but that wouldn't make sense if the errors obj only changes based on the single input's 'onChange' and not when the input it depends on changes
I thought I saw somewhere in the docs that showed this single input at a time could be turned off? Or maybe I'm imagining since I can't find it right now.

**Reply by bluebill1049** [reply]
is this what you refer to `onChange={() => trigger(['input1', 'input2'])}` ?

**Reply by chornbec** [reply]
Yes, I believe so. If input1 is required because input2 is "hello", an onChange of input2 would trigger re-validation in case input1 is still required.

**Reply by bluebill1049** [reply]
you probably want to write some logic inside the onChange then.
